integrations/amplitude/lib/index.js

```diff
@@ -286,8 +294,28 @@ Amplitude.prototype.orderCompleted = function(track) {
   var trackRevenuePerProduct = this.options.trackRevenuePerProduct;
   var revenueType = track.proxy('properties.revenueType');
   var revenue = track.revenue();
-
-  // Amplitude does not allow arrays of objects to as properties of events.
+  var trackProductsOnce = this.options.trackProductsOnce;
+
+  if (trackProductsOnce) {
+    var allProducts = [];
+    // products is object
+    var productKeys = Object.keys(products);
+    for (var index = 0; index < productKeys.length; index++) {
+      var eachProduct = new Track({ properties: products[index] });
+
+      allProducts.push({
```

Comment:
I'm wondering if we can use `mapRevenueAttributes()` to retrieve all the properties of each product instead of making it again. What do you think?

Reply:
created similar method as of **mapRevenueAttributes** as there is some attribute difference.